squirrels/_api_routes/auth.py

```
@@ -149,93 +176,116 @@ async def provider_callback(request: Request, provider_name: str):
-            api_key, _ = self.authenticator.create_access_token(user, expiry_minutes=body.expiry_minutes, title=body.title)
-            return rm.ApiKeyResponse(api_key=api_key)
-        
-        @auth_router.get(api_key_path, description="Get all API keys with title for the current user", tags=["Authentication"])
-        async def get_all_api_keys(user: RegisteredUser | GuestUser = Depends(self.get_current_user)):
-            if isinstance(user, GuestUser):
-                raise InvalidInputError(401, "invalid_authorization_token", "Invalid authorization token, cannot get API keys")
-            return self.authenticator.get_all_api_keys(user.username)
-        
-        revoke_api_key_path = '/api-keys/{key_id}'
+            @auth_router.get(api_key_path, description="Get all API keys with title for the current user", tags=["Authentication"])
+            async def get_all_api_keys(user: RegisteredUser | GuestUser = Depends(self.get_current_user)):
```

Ruff (0.14.13)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)